### DISCUSSION TITLE
Link specific tag pages on OSM Wiki

### DISCUSSION THREAD

**Discussion by matkoniecz** [streetcomplete/StreetComplete#3442] [Q&A] [answered]
I think that it may be a good idea to give ability to reach description of specific tags used by SC from within app.

Yes, OSM Wiki is linked in achievements but it is not really clear that tags are described and what tags actually are.

Currently tags are visible just in undo menu, but maybe it would be a good idea to make them linkable there?

Another related idea where it would make sense to show this is "Feature: Show the list of edits in the profile" #2907

**Comment by kmpoppe** [comment]
Very much in favour of this. Adding a single Key or Key/Value to any quest that could be opened in the Phone's language (existence can be checked using the MediaWiki API with one or two requests) would be easy, just the GUI would be a challenge because there isn't that much unused space in the quest popup, is there?

**Reply by matkoniecz** [reply]
> existence can be checked using the MediaWiki API with one or two requests

I would rather build list of existing pages and update it on release (like OSM website is doing) instead of runtime checks that would add lag and keep hammering OSM Wiki server.

> just the GUI would be a challenge because there isn't that much unused space in the quest popup, is there?

Definitely not in quest UI. I thought about linking tags in undo menu.

There is link already in bubble quest view but "Documentation" is not really too inviting...

**Reply by riQQ** [reply]
> I would rather build list of existing pages and update it on release (like OSM website is doing) instead of runtime checks that would add lag and keep hammering OSM Wiki server.

Agreed

> Definitely not in quest UI.

Why not? I know that it's already filled with stuff, but that's exactly the place where I would have expected to find such a link and didn't find one when I started using StreetComplete. The links in the bubbles in the profile is way too hidden to be discoverable.

**Reply by matkoniecz** [reply]
> Why not? I know that it's already filled with stuff

That is the exact reason. There is no space to fit anything more there.

**Reply by riQQ** [reply]
See https://github.com/streetcomplete/StreetComplete/issues/1913#issuecomment-648845579 for an example of how it could work.

**Reply by peternewman** [reply]
> > existence can be checked using the MediaWiki API with one or two requests
> 
> I would rather build list of existing pages and update it on release (like OSM website is doing) instead of runtime checks that would add lag and keep hammering OSM Wiki server.

Do you mean a list of every page on the wiki at that time?

You could use #2754 (if I ever finish it), to narrow the list for just those which we tag.

**Comment by westnordost** [comment]
> OSM Wiki is linked in achievements

You mean in the quest type statistics.

Anyway, maybe what could be done is to make the tags displayed in the undo-dialog clickable? Though oftentimes one will land on a "this page doesn't exist yet" page.

**Reply by matkoniecz** [reply]
> Though oftentimes one will land on a "this page doesn't exist yet" page.

I though about storing list of existing pages, similar to how openstreetmap.org is doing this.

See https://github.com/openstreetmap/openstreetmap-website/pull/3294

Maybe even outright mirror https://github.com/matkoniecz/openstreetmap-website/blob/master/config/wiki_pages.yml ? Rather than building own? On the other hand it is 1MB extra so maybe something smarter would be better...

**Comment by matkoniecz** [comment]
Additional question: what should be linked if user is using say German language? German translation page or the original English?

Personally I am tempted to always link English, as one may easily switch to translations if present and translations on wiki are typically inferior to English version (which is often missing info or misleading or benefit in other way from edits)

@westnordost How long full release build works? Is it taking long time or is it something running quickly? How much time can be added without making it a problem for you?

Or maybe such link-presence check can be run in parallel wit other build task and would not make release build time longer?

I am asking as this may be adding extra build time - for example updating https://github.com/openstreetmap/openstreetmap-website/blob/master/config/wiki_pages.yml takes about two hours.

Overall I am tempted to rely on https://github.com/openstreetmap/openstreetmap-website/blob/master/config/wiki_pages.yml as source for presence of OSM wiki pages, to avoid rechecking their presence during build time as it takes noticeable time (about 10-20 minutes to check presence of English pages, though I have not tried smarter API calls or parallel workers).

**Reply by mnalis** [reply]
I would also prefer linking to English version, for reasons mentioned

**Comment by matkoniecz** [comment, marked as answer]
https://github.com/streetcomplete/StreetComplete/pull/4419 - done, will be in v48
